includes/core/stateManager/StateManager.hpp

```diff
@@ -1,16 +1,37 @@
 #pragma once
+#include "State.hpp"
+
+#include <string>
+#include <map>
+#include <stack>
+#include <iostream>
+#include <stdexcept>
+#include <functional>
+#include <memory>
+#include <cstdint>
 
 namespace Yuna
 {
 	namespace Core
 	{
+		typedef std::function<std::unique_ptr<State>(uint8_t)> StateFactory;
 		class StateManager
 		{
 		private:
-			/* data */
+			std::stack<std::unique_ptr<State>>		mStateStack;
+
+			StateFactory			mStateFactory;
```

Comment:
The State factory is the function that will help the state manager know which state to add the the state stack
(See sandbox for example in game.cpp)